src/helpers/buffers/unordered_receiver.rs

```diff
@@ -226,6 +250,62 @@ where
             }
         }
     }
+
+    fn get_waiting_messages(&self) ->LoggingRanges {
+        let mut response = vec![self.next];
+        for i in self.next + 1 ..self.next + self.wakers.len()+1 as usize {
+            if let Some(_) = self.wakers[i % self.wakers.len()] {
+                response.push(i);
+            }
+        }
+        LoggingRanges::from(&response)
+    }
+}
+
+#[cfg(debug_assertions)]
+struct IdleTrackOperatingState<S, C>
+where
+    S: Stream<Item = C>,
+    C: AsRef<[u8]>, {
+        state: OperatingState<S, C>,
+        current_next: usize,
+    }
+
+#[cfg(debug_assertions)]
+impl<S, C> IdleTrackOperatingState<S, C> where
+    S: Stream<Item = C> + Send,
+    C: AsRef<[u8]>, {
+     fn new(stream: Pin<Box<S>>, next: usize,spare: Spare,wakers: Vec<Option<Waker>>,overflow_wakers: Vec<Waker>,
```

Comment:
just implement `From<S> for IdleTrackOperatingState` so you don't need to pass stuff through the constructor